features/staging_commands.feature

```diff
@@ -542,3 +542,31 @@ Feature: Enable command behaviour when attached to an UA staging subscription
            | release | fips-apt-source                                                        |
            | xenial  | https://esm.staging.ubuntu.com/fips-updates/ubuntu xenial-updates/main |
            | bionic  | https://esm.staging.ubuntu.com/fips-updates/ubuntu bionic-updates/main |
+
+
+    @series.lts
+    @uses.config.machine_type.lxd.vm
+    Scenario Outline: Commands when cloud init didn't run properly
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I delete the file `/run/cloud-init/instance-data.json`
+        And I attach `contract_token_staging` with sudo
+        Then stderr matches regexp:
+        """
+        Warning: cloud-init is present but did not run. Assuming not on cloud. If you are on a cloud, you may experience unreliable behavior.
+        """
+        When I run `ua status` with sudo
+        Then stderr matches regexp:
```

Comment:
Instead of having that warning appear on the beginning of status, we could add it as a notice and just delete after we print the status:

We should also consider this mechanism for the other commands as well

Reply:
Good idea to use notices for this. Why would we delete the notice after printing the status? Instead, I would expect we'd delete the notice when this function successfully finds a cloud or definitely returns None (any other code path that doesn't result in this warning).

Reply:
@orndorffgrant maybe this is a minor thing, but I thought about avoiding have that particular notice be in the `json|yaml` output.

But maybe it is not a bad idea to have it there

Reply:
I think it makes sense that the notice will stay until the problem is solved.